README.markdown

```diff
@@ -21,67 +21,47 @@ It should work on any system that supports Google's [Cloud SDK](https://develope
-associated with a specified set of credential placeholder path and project
-identifier.  The element type should be set to 'gce' and the url should
-contain both a file path and the name of the project in the format below:
+#### Setup a Puppet Agent inside of your Google Cloud project
 
-    #/etc/puppet/device.conf
-    [my_project1]
-      type gce
-      url [/dev/null]:project_id
+If you would like to use a virtual machine inside of your project as your Puppet Agent, setup is simple.  Create the instance manually, either on the [Developers Console](https://console.developers.google.com/) or via the `gcloud` command-line interface, making sure to enable the `compute-rw` scope for your instance.
 
-Note that this version of the gce_compute module sets the file path to
-`/dev/null`.  This is a placeholder value that will be used to reference
-a credentials file in a newer release of this module.  For now, setting
-the value to `/dev/null` is a working solution as long as you have previously
-set up Cloud SDK with the `gcloud auth login` command.
+- In the Developers Console, create an instance via `Compute > Compute Engine > VM instances > New instance`, show the security options, and select "Read Write" under `Project Access > Compute`.
+- In `gcloud`, just use `gcloud compute instances create` with the `--scopes compute-rw` flag.
 
-### Multiple Projects
+Once you've setup your instance with the `compute-rw` scope, you don't need do anything else: `gcloud` comes preinstalled on the VM, and the instance is able to read and write resources within its project.
 
-You can use multiple cloud projects by making the appropriate entries in your
-`device.conf` file and adjusting the Cloud SDK settings.  For each project,
-you'll first need to create authorize each project with:
+#### Setup a Puppet Agent outisde of your Google Cloud project
 
-    gcloud config set account ANOTHER_ACCOUNT_NAME
-    gcloud auth login
+If you would like to use a machine outside of your project as your Puppet Agent, you'll need to [install and authenticate gcloud](https://cloud.google.com/sdk/).
 
-Once all of your projects have been authorized, you can toggle which project
-will be used prior to invoking `puppet apply` by using:
+### Install Puppet and this module
 
-    gcloud config set project PROJECT
+You'll now want to [install Puppet](https://docs.puppetlabs.com/guides/install_puppet/pre_install.html) on your Puppet Agent.  Once you've installed Puppet, do
 
-The example below show how multiple certificate names can be used to represent
-multiple projects in GCE.
+```bash
+$ puppet module install puppetlabs-gce_compute
+```
 
-    #/etc/puppet/device.conf
-    [certname1]
-      type gce
-      url [/dev/null]:group:my_project1
-    [certname2]
-      type gce
-      url [/dev/null]:group:my_project2
+At this point, you should be ready to go!
 
 ## Quick Start with Puppet Enterprise
 
+*Puppet Enterprise ships with `facter` and when run will attempt to read the
```

Comment:
Puppet also installs facter, this shouldn't differ between PE and the OSS release. Puppet should certainly work without needing root permissions, could you provide information here about how to recreate the problem you ran into - I'm not sure this should be required in the README.